grammar.y

```diff
@@ -1004,10 +1004,20 @@ column_constraints_opt:
 column_constraints:
   column_constraint
   {
+    if _, ok := $1.(*ColumnConstraintPrimaryKey); ok {
```

Comment:
Adds the check in the column definition as well.

Also, changed the variable `createStmtHasPrimaryKey` from global to local `yylex.(*Lexer).createStmtHasPrimaryKey` to avoid data races. It's kind of an ugly pattern but looks like it's the only way to do it with go yacc.

Reply:
Cool :+1: